src/platform/test.rs

```diff
@@ -414,6 +414,55 @@ fn receiver_set() {
 }
 
 #[test]
+#[cfg(not(any(feature = "force-inprocess", target_os = "windows", target_os = "android")))]
+fn receiver_set_eintr() {
+    let mut stat: i32 = 0;
+    let (tx0, rx0) = platform::channel().unwrap();
+    let (tx1, rx1) = platform::channel().unwrap();
+    let mut rx_set = OsIpcReceiverSet::new().unwrap();
+    let mut result_set = OsIpcReceiverSet::new().unwrap();
+    rx_set.add(rx1).unwrap();
+    let child_pid = unsafe {
+        fork(|| {
+            // Let the parent know we're ready
+            tx0.send(b"B", vec![], vec![]).unwrap();
+            // Send the result of the select back to the parent
+            match rx_set.select() {
+                Ok(_) => tx0.send(b"\x00", vec![], vec![]).unwrap(),
+                Err(e) => {
+                    let err = format!("{:?}", e);
+                    tx0.send(err.into_bytes().as_slice(), vec![], vec![]).unwrap();
+                }
+            }
+        })
+    };
+    // Wait until the child is ready
+    rx0.recv().unwrap();
+    unsafe {
+        libc::kill(child_pid, libc::SIGSTOP);
+        thread::sleep(Duration::from_millis(42));
+        libc::kill(child_pid, libc::SIGCONT);
+    }
+    // The interrupt shouldn't affect the following send
+    tx1.send(b"B", vec![], vec![]).unwrap();
+    // Receive the results from the child and panic if the the
+    // returned value from select was an error.
+    result_set.add(rx0).unwrap();
+    for result in result_set.select().unwrap() {
```

Comment:
I don't see a need to use a set here: this channel is only for getting back the result; the actual select test is done when sending to the child...